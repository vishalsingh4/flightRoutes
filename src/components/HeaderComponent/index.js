// css import
import './index.scss';

// images import
import logo from 'assets/logo.png';
import avatar from 'assets/avatar.png';

const HeaderComponent = () => {
    return (
        <nav className='navbar navbar-expand-lg bg-dark d-flex align-items-center p-0'>
            <img className='logo' src={logo} alt='logo-preview' width='75' height='75' />
            <div className='manage-section d-flex justify-content-around align-items-center color-white fs-15 fw-600 w-50'>
                <span className='link' tabIndex='0' aria-label='book'>BOOK</span>
                <span className='link' tabIndex='0' aria-label='manage'>MANAGE</span>
                <span className='link' tabIndex='0' aria-label='experience'>EXPERIENCE</span>
                <span className='link' tabIndex='0' aria-label='where we fly'>WHERE WE FLY</span>
                <span className='link' tabIndex='0' aria-label='loyalty'>LOYALTY</span>
                <span className='link' tabIndex='0' aria-label='help'>HELP</span>
            </div>
            <div className='user position-absolute'>
                <img src={avatar} alt='user-avatar' width='30' height='30' />
                <span className='logged-user color-white fs-15 fw-600' tabIndex='0' aria-label='katy pearson'>Katy Pearson</span>
            </div>
        </nav>
    )
};

export default HeaderComponent;