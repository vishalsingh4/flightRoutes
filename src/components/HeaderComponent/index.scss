  /* Extra small devices (phones, 600px and down) */
  @media only screen and (max-width: 600px) {
    .header-section {
        nav {
            flex-direction: column;
            .logo {
                width: 50px;
                height: 50px;
                margin-bottom: 7px;
            }

            .manage-section {
                width: 100% !important;
                font-size: 9px;
            }

            .user {
                img {
                    width: 20px;
                    height: 20px;
                }
                .logged-user {
                    font-size: 9px;
                }
            }
        }
    }
  }

/* Small devices (portrait tablets and large phones, 600px and up) */
  @media only screen and (min-width: 600px) and (max-width: 768px) {
    .header-section {
      nav {
          flex-direction: column;
          .logo {
              width: 60px;
              height: 60px;
              margin-bottom: 10px;
          }

          .manage-section {
              width: 100% !important;
              font-size: 12px;
          }

          .user {
              img {
                  width: 25px;
                  height: 25px;
              }
              .logged-user {
                  font-size: 10px;
              }
          }
      }
    }
  }

/* Medium devices (landscape tablets, 768px and up) */
  @media only screen and (min-width: 768px) and (max-width: 992px){
    .header-section {
      nav {
          flex-direction: column;
          .logo {
              width: 70px;
              height: 70px;
              margin-bottom: 12px;
          }

          .manage-section {
              width: 100% !important;
              font-size: 15px;
          }

          .user {
              img {
                  width: 28px;
                  height: 28px;
              }
              .logged-user {
                  font-size: 12px;
              }
          }
      }
    }
  } 