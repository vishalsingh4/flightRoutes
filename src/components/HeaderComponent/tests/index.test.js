import React from 'react';
import { shallow } from 'enzyme';
import HeaderComponent from '..';

import logo from 'assets/logo.png';
import avatar from 'assets/avatar.png';

describe("Header Component", () => {
    const wrapper = shallow(<HeaderComponent />);

    it('should render the Header Component correctly - snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('renders component', () => {
        expect(wrapper.exists()).toBe(true);
    });
    it('renders the header logo', () => {
        expect(wrapper.find('img').first().prop('src')).toEqual(logo);
    });
    it('renders the user avatar', () => {
        expect(wrapper.find('img').last().prop('src')).toEqual(avatar);
    });
    it('renders the links', () => {
        expect(wrapper.find('.link').at(0).text()).toEqual('BOOK');
        expect(wrapper.find('.link').at(1).text()).toEqual('MANAGE');
        expect(wrapper.find('.link').at(2).text()).toEqual('EXPERIENCE');
        expect(wrapper.find('.link').at(3).text()).toEqual('WHERE WE FLY');
        expect(wrapper.find('.link').at(4).text()).toEqual('LOYALTY');
        expect(wrapper.find('.link').at(5).text()).toEqual('HELP');
    });
    it('render logged in user', () => {
        expect(wrapper.find('.logged-user').text()).toEqual('Katy Pearson');
    });
})