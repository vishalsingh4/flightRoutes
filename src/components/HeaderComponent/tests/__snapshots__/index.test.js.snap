// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render the Header Component correctly - snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg bg-dark d-flex align-items-center p-0"
>
  <img
    alt="logo-preview"
    className="logo"
    height="75"
    src="logo.png"
    width="75"
  />
  <div
    className="manage-section d-flex justify-content-around align-items-center color-white fs-15 fw-600 w-50"
  >
    <span
      aria-label="book"
      className="link"
      tabIndex="0"
    >
      BOOK
    </span>
    <span
      aria-label="manage"
      className="link"
      tabIndex="0"
    >
      MANAGE
    </span>
    <span
      aria-label="experience"
      className="link"
      tabIndex="0"
    >
      EXPERIENCE
    </span>
    <span
      aria-label="where we fly"
      className="link"
      tabIndex="0"
    >
      WHERE WE FLY
    </span>
    <span
      aria-label="loyalty"
      className="link"
      tabIndex="0"
    >
      LOYALTY
    </span>
    <span
      aria-label="help"
      className="link"
      tabIndex="0"
    >
      HELP
    </span>
  </div>
  <div
    className="user position-absolute"
  >
    <img
      alt="user-avatar"
      height="30"
      src="avatar.png"
      width="30"
    />
    <span
      aria-label="katy pearson"
      className="logged-user color-white fs-15 fw-600"
      tabIndex="0"
    >
      Katy Pearson
    </span>
  </div>
</nav>
`;
